lib/_http_agent.js

```diff
@@ -340,6 +338,11 @@ Agent.prototype.createSocket = function createSocket(req, options, cb) {
     oncreate(null, newSocket);
 };
 
+function normalizeServerName(options, req) {
+  if (!options.servername && options.servername !== '')
+    options.servername = calculateServerName(options, req);
+}
+
```

Comment:
Should we avoid mutating an argument in a function?

Reply:
I agree it would be good to refactor that away. But can happen in a follow up PR.